libtiledbvcf/src/write/writer.cc

```diff
@@ -608,11 +614,27 @@ std::pair<uint64_t, uint64_t> Writer::ingest_samples(
         if (worker->records_buffered() > 0) {
           worker->buffers().set_buffers(
               query_.get(), dataset_->metadata().version);
-          auto st = query_->submit();
-          if (st != Query::Status::COMPLETE)
-            throw std::runtime_error(
-                "Error submitting TileDB write query; unexpected query "
-                "status.");
+          query_->submit();
+
+          if (remote_) {
+            if (query_->query_status() == Query::Status::FAILED) {
+              LOG_FATAL(
+                  "Error submitting TileDB write query; unexpected query "
+                  "status: {}",
+                  // TODO: fix for SC-26607
+                  static_cast<int>(query_->query_status()));
+            }
+            worker->buffers().clear_query_buffers(
+                query_.get(), dataset_->metadata().version);
+          } else {
```

Comment:
Can we rewrite this status check so it doesn't depend on tracking is the array is remote?